src/main/java/com/tstepnik/planner/controller/RegistrationController.java

```diff
@@ -17,19 +19,21 @@
 public class RegistrationController {
 
     private final RegistrationService registrationService;
+    private final RegisterUserMapper mapper;
 
     @Autowired
-    public RegistrationController(RegistrationService registrationService) {
+    public RegistrationController(RegistrationService registrationService, RegisterUserMapper mapper) {
         this.registrationService = registrationService;
+        this.mapper = mapper;
     }
 
     @PostMapping("/register")
-    public ResponseEntity<User> registerUser(@RequestBody User user) {
-        Optional<User> registerUser = Optional.ofNullable(registrationService.register(user));
+    public ResponseEntity<RegisterUserDto> registerUser(@RequestBody RegisterUserDto user) {
```

Comment:
This RegisterUserDTO is a little bit confusing. I will probably leave you other comments on this later.